solidity/ts/tests/test.ts

```diff
@@ -1,58 +1,448 @@
-	if (deployerBytecode === '0x60003681823780368234f58015156014578182fd5b80825250506014600cf3') return
-	const ethSendHash = await client.sendTransaction({ to: '0x4c8d290a1b368ac4728d83a9e8321fc3af2b39b1', amount: 10000000000000000n })
-	await client.waitForTransactionReceipt({ hash: ethSendHash })
-	const deployHash = await client.sendRawTransaction({ serializedTransaction: '0xf87e8085174876e800830186a08080ad601f80600e600039806000f350fe60003681823780368234f58015156014578182fd5b80825250506014600cf31ba02222222222222222222222222222222222222222222222222222222222222222a02222222222222222222222222222222222222222222222222222222222222222' })
-	await client.waitForTransactionReceipt({ hash: deployHash })
-}
-
-export const deployAugurConstantProductMarketContract = async (client: WriteClient) => {
-	const contractLocation = './artifacts/AugurConstantProductMarket.json'
-	const contracts = ContractArtifact.parse(JSON.parse(await fs.readFile(contractLocation, 'utf8')))
-	const contract = contracts.contracts['AugurConstantProductMarket.sol'].AugurConstantProduct.evm.bytecode.object
-	await ensureProxyDeployerDeployed(client)
-	const bytecode: `0x${ string }` = `0x${ contract }`
-	const transaction = { to: proxyDeployerAddress, data: bytecode } as const
-	const hash = await client.sendTransaction(transaction)
-	await client.waitForTransactionReceipt({ hash })
-}
-
-
-const vitalik = 0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045n
-
-const canDeployContract = async () => {
-	const client = createWriteClient(getMockedEthSimulateWindowEthereum(), vitalik, 0)
+import { createWriteClient } from '../testsuite/simulator/utils/viem.js'
+import { RICH_ADDRESS_1, RICH_ADDRESS_2, RICH_ADDRESS_3, RICH_ADDRESS_4 } from '../testsuite/simulator/utils/constants.js'
+import { deployAugurConstantProductMarketContract, isAugurConstantProductMarketDeployed, approveDai, getDaiAllowance, addLiquidity, getPoolLiquidityBalance, removeLiquidity, getDaiBalance, getReportingFee, getShareBalances, enterPosition, getAugurConstantProductMarketAddress, expectedSharesAfterSwap, exitPosition, getShareToken, setERC1155Approval, swap, getPoolSupply, getPoolConstant, getNoYesShareBalances } from '../testsuite/simulator/utils/utilities.js'
+import assert from 'node:assert'
+
+const numTicks = 1000n;
+
+test('canDeployContract', async () => {
+	const client = createWriteClient(getMockedEthSimulateWindowEthereum(), RICH_ADDRESS_1, 0)
+	await deployAugurConstantProductMarketContract(client)
+	const isDeployed = await isAugurConstantProductMarketDeployed(client)
+	assert.ok(isDeployed, `Not Deployed!`)
+})
+
+test('canAddAndRemoveLiquidity', async () => {
+	const client = createWriteClient(getMockedEthSimulateWindowEthereum(), RICH_ADDRESS_1, 0)
+	await deployAugurConstantProductMarketContract(client)
+
+	// Approve Dai for ACPM
+	await approveDai(client)
+	const allowance = await getDaiAllowance(client)
+	assert.notEqual(allowance, 0n, `Approve failed`)
```

Comment:
```suggestion
	assert.notEqual(allowance, 0n, 'Approve failed')
```

Reply:
In general I like using ``` even if it currently has no templating so that if it needs to be added for debugging or later display its always setup correctly. Its also consistent with other error message formatting then.